homotopy-web/src/model.rs

```diff
@@ -153,18 +140,14 @@ impl State {
                 let svg = svg_element.inner_html();
 
                 // We must now pull all the relevant stylesheets that are needed in the SVG.
-                // Failure to do so gives a fully-black SVG. Since getting direct access to the SVG
-                // stylesheet render seems too complicated in the current state, as SignatureStylesheet
-                // is in a private module, we use again the trick of pulling everything from the DOM.
-                // Note that this solution would be much simpler if we opted for saving the
-                // rendered SVG *somewhere*, then just get it from there.
-                let style_element = web_sys::window()
-                    .expect("no window")
-                    .document()
-                    .expect("no document")
-                    .get_element_by_id("signature__stylesheet")
-                    .expect("no stylesheet in document");
-                let stylesheet = style_element.outer_html();
+                // Failure to do so gives a fully-black SVG. We can generate the stylesheet in the
+                // same way the `SignatureStylesheet` struct does.
+                // We also strip the styles of whitespace since it is unneeded.
+                let stylesheet = {
+                    let mut inner_stylesheet = svg::stylesheet(&signature, "generator");
+                    inner_stylesheet.retain(|c| !c.is_whitespace());
```

Comment:
The whitespaces might be nice for readability. It might be worth keeping them for the export (since people might want to look at the svg code if they downloaded it).

Reply:
Yeah, I agree though the rest of the svg code is all a single line. Perhaps we could also make _that_ multiline as well for the same reason.

Reply:
Yes good point, I will fix that as part of #609.